src/app/campaign/store/actions/campaign-action.creator.ts

```diff
@@ -85,25 +88,39 @@ export class CampaignSaveFailure implements Action {
 export class CampaignAddFlight implements Action {
   readonly type = ActionTypes.CAMPAIGN_ADD_FLIGHT;
 
-  constructor(public payload: { campaignId: number | string }) {}
-}
-
```

Comment:
Removed the additional action dispatch from the effects for add and dup flight. The flightId/timestamps are generated in the action creators. Now the effect just handles router navigation without an extra action dispatch, and the reducer responds to the original action.